src/apps/concepts/pages/ViewConceptsPage.tsx

```diff
@@ -111,9 +113,13 @@ const ViewConceptsPage: React.FC<Props> = ({
   profile,
   usersOrgs,
   containerType,
+  viewDictConcepts,
   addConceptsToDictionary,
   removeConceptsFromDictionary,
 }) => {
+  const dictionaryConcepts = dictionary?.references.map(r => r.expression);
+  const modifiedConcepts = concepts?.map(c => includes(dictionaryConcepts, c.version_url) ? {...c, added: true} : {...c});
```

Comment:
I think we better to do this mapping in the `mapStateToProps` function. That way we're working with Redux instead of fighting it.